### PR TITLE
Decode benchmark

### PR DESCRIPTION
**Pull request by elias-orijtech**
CC @odeke-em.

### CODE REVIEW

**Review by odeke-em — APPROVED** [review]
LGTM, thank you @elias-orijtech!

**Review by liamsi — APPROVED** [review]
👍🏼

File: options.go [hunk 1]
```diff
@@ -205,3 +206,11 @@ func WithCustomMatrix(customMatrix [][]byte) Option {
 		o.customMatrix = customMatrix
 	}
 }
+
+// WithLeopard will always use leopard for encoding.
+// Note that Leopard places certain restrictions on use see other documentation.
+func WithLeopard(enabled bool) Option {
```
Comment (klauspost) [inline options.go:212]:
```suggestion
func WithLeopardGF16(enabled bool) Option {
```

File: options.go [hunk 2]
```diff
@@ -205,3 +206,11 @@ func WithCustomMatrix(customMatrix [][]byte) Option {
 		o.customMatrix = customMatrix
 	}
 }
+
+// WithLeopard will always use leopard for encoding.
+// Note that Leopard places certain restrictions on use see other documentation.
```
Comment (klauspost) [inline options.go:211]:
```suggestion
// WithLeopardGF16 will always use leopard GF16 for encoding.
// This is not compatible with Leopard GF8 output for shards < 256.
// Note that Leopard places certain restrictions on use see other documentation.
```

**Review by klauspost — CHANGES_REQUESTED** [review]
Maybe this is clear enough?

File: options.go (hunk 2 above)
Comment (elias-orijtech) [reply options.go:211]: Nit: "for shards < 256" is slightly misleading because FF8 doesn't support shards < 256. Otherwise, don't you mean "<= 256"?

**Comment by elias-orijtech** [comment]
> Maybe this is clear enough?

Will there ever be a WithLeopardGF8? If not, the name is ugly and the FF16 part could be a documented detail.

It looks like the only difference between leopard GF16 and GF8 is the decode speed. Can GF16 inversion matrices be cached for, say, shards <= 256 to bridge the gap?

**Comment by klauspost** [comment]
> Will there ever be a WithLeopardGF8? If not, the name is ugly and the FF16 part could be a documented detail.

I think it would be reasonable to leave that option open. It does provide a nice speedup.

I agree it is ugly, but I think that also makes it pretty clear that there is something "special" about it, and it isn't leopard compatible.

**Comment by klauspost** [comment]
I merges and will send my changes as separate PR.

**Pull request merged**